presto-main/src/test/java/com/facebook/presto/sql/analyzer/TestAnalyzer.java

```diff
@@ -317,6 +327,13 @@ public void testOrderByMustAppearInSelectWithDistinct()
     }
 
     @Test
+    public void testNonDeterministicOrderBy()
+    {
+        analyze("SELECT DISTINCT random() as b FROM t1 ORDER BY b");
+        assertFails(NONDETERMINISTIC_ORDER_BY_EXPRESSION, "SELECT DISTINCT random() FROM t1 ORDER BY random()");
```

Comment:
With distinct ORDER BY expressions must be one of SELECT DISTINCT expressions. You first execute distinct (which reorders rows) and then do sort (only SELECT DISTINCT expressions are available after distinct)